### PR TITLE
feat(device-scale): Migrate device-scale device service to EdgeX v3.0

### PR DESCRIPTION
**Pull request by seanohair22**
## PR Checklist

- [x] Added label to the Pull Request for easier discoverability and search
- [x] Commit Message meets guidelines as indicated in the URL https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md
- [x] Every commit is a single defect fix and does not mix feature addition or changes
- [ ] Unit Tests have been added for new changes
- [ ] Updated Documentation as relevant to the changes
- [ ] All commented code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with repository license and clearly outlined the added dependency.
- [ ] PR change contains code related to security
- [ ] PR introduces changes that breaks compatibility with other modules (If YES, please provide details below)

## Issue this PR will close

close: #**74**
closes: #74 

## Test Instructions if applicable

- Run `make device-scale` 
- Comment out `event-reconciler`, `product-lookup` and `loss-detector` in `docker-compose.loss-detection.yml`
- Run `make run-portainer run-base`
- Navigate to `localhost:9000`
- Check `device-scale` portainer logs 
    - Note: `device-scale` will have a warning/error about not finding the serial port and/or the scale is not connected. (but that is okay)

### CODE REVIEW

File: rtsf-at-checkout-device-scale/Dockerfile [hunk 1]
```diff
@@ -19,18 +19,18 @@ RUN go mod download
 COPY . .
 
 RUN go mod tidy
-RUN go build -tags no_zmq -o device-scale cmd/main.go
+RUN CGO_ENABLED=0 go build -tags no_zmq -o device-scale cmd/main.go
```
Comment (lenny-intel) [inline rtsf-at-checkout-device-scale/Dockerfile:22]: Tag no longer exists now that ZMQ has been totally removed
```suggestion
RUN CGO_ENABLED=0 go build -o device-scale cmd/main.go
```

File: rtsf-at-checkout-device-scale/Dockerfile [hunk 2]
```diff
@@ -19,18 +19,18 @@ RUN go mod download
...
 WORKDIR /
 COPY --from=builder /device-scale/device-scale /device-scale
-COPY --from=builder /device-scale/cmd/res/configuration.toml /res/configuration.toml
+COPY --from=builder /device-scale/cmd/res/configuration.yaml /res/configuration.yaml
 COPY --from=builder /device-scale/cmd/res/profiles/cas-scale.yaml /res/profiles/cas-scale.yaml
-COPY --from=builder /device-scale/cmd/res/devices/device-list.toml /res/devices/device-list.toml
+COPY --from=builder /device-scale/cmd/res/devices/device-list.yaml /res/devices/device-list.yaml
 
-CMD [ "/device-scale","--cp=consul://edgex-core-consul:8500","--registry","--confdir=/res"]
+CMD [ "/device-scale","--cp=consul://edgex-core-consul:8500"]
```
Comment (lenny-intel) [inline rtsf-at-checkout-device-scale/Dockerfile:36]: Still can have `--registry`
```suggestion
CMD [ "/device-scale","--cp=consul://edgex-core-consul:8500","--registry"]
```

File: rtsf-at-checkout-device-scale/driver/scaleDriver.go [hunk 3]
```diff
@@ -205,3 +200,46 @@ func (drv *ScaleDriver) RemoveDevice(deviceName string, protocols map[string]mod
...
+
+// Discover is a callback function that is invoked
+// by the SDK but should never be called
+func (drv *ScaleDriver) Discover() error {
+
+	return errors.New("discovery not implemented")
+}
+
+// Validate is a callback function that is invoked
+// by the SDK but should never be called.
```
Comment (lenny-intel) [inline rtsf-at-checkout-device-scale/driver/scaleDriver.go:212]:
```suggestion
```

**Review by lenny-intel — CHANGES_REQUESTED** [review]
Please update the CopyRight year on all the files you modified.

**Review by presatish — CHANGES_REQUESTED** [review]
Looks like some unit tests are failing.

**Review by lenny-intel — APPROVED** [review]
LGTM

**Review by marcpfuller — APPROVED** [review]
LGTM

**Review by brian-intel — APPROVED** [review]

**Pull request merged**